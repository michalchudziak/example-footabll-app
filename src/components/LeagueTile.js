/* @flow */
import React from 'react';
import { Dimensions, StyleSheet } from 'react-native';
import styled from 'styled-components/native';

type Props = {
  name: string,
  country: string,
  color?: string,
  onClick: () => void,
};

const Container = styled.TouchableOpacity`
  flex: 1;
  width: ${Dimensions.get('window').width};
  border-bottom-color: #000000;
  border-bottom-width: ${StyleSheet.hairlineWidth};
  justify-content: center;
  align-items: center;
  background-color: ${props => props.color || 'transparent'}
`;

const Title = styled.Text`
  font-size: 18;
`;

const Subtitle = styled.Text`
  font-size: 12;
`;

export default function LeagueTile({ name, country, color, onClick }: Props) {
  return (
    <Container color={color} onPress={onClick}>
      <Title>{name}</Title>
      <Subtitle>{country}</Subtitle>
    </Container>
  );
}
