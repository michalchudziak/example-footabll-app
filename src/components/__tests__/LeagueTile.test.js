/* @flow */
import React from 'react';
import renderer from 'react-test-renderer';
import LeagueTile from '../LeagueTile';

describe('<LeagueTile />', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<LeagueTile name="League" country="Poland" onClick={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
