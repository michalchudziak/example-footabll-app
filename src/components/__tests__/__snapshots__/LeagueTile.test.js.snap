// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeagueTile /> should render correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderBottomColor": "#000000",
      "borderBottomWidth": 0.5,
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
      "opacity": 1,
      "width": 750,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 18,
        },
        undefined,
      ]
    }
  >
    League
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 12,
        },
        undefined,
      ]
    }
  >
    Poland
  </Text>
</View>
`;
